File: client/src/model/read/Story.ts
import {requireString} from "../Deserialise";


export interface Story {
  name: string;
  urlname: string;

  description: string;
};


export function asStory(obj: any): Story {
  return {
    name: requireString(obj["name"]),
    urlname: requireString(obj["urlname"]),
    description: requireString(obj["description"])
  };
}

File: service/GoodNight.Service.Domain/Model/Read/Transfer/Story.cs
namespace GoodNight.Service.Domain.Model.Read.Transfer
{
  /// <summary>
  /// A short summary of a story. This is mostly used for serialisation and
  /// transfer to the client, as the client never sees the whole story.
  /// </summary>
  public record Story(
    string Name,
    string Urlname,
    string? Icon,
    string Description);
}
